quaddtype/numpy_quaddtype/src/dtype.c

```diff
@@ -0,0 +1,261 @@
+#include <Python.h>
+#include <sleef.h>
+#include <sleefquad.h>
+
+#define PY_ARRAY_UNIQUE_SYMBOL QuadPrecType_ARRAY_API
+#define PY_UFUNC_UNIQUE_SYMBOL QuadPrecType_UFUNC_API
+#define NPY_NO_DEPRECATED_API NPY_2_0_API_VERSION
+#define NPY_TARGET_VERSION NPY_2_0_API_VERSION
+#define NO_IMPORT_ARRAY
+#define NO_IMPORT_UFUNC
+#include "numpy/arrayobject.h"
+#include "numpy/ndarraytypes.h"
+#include "numpy/dtype_api.h"
+
+#include "scalar.h"
+#include "casts.h"
+#include "dtype.h"
+#include "dragon4.h"
+
+static inline int
+quad_load(void *x, char *data_ptr, QuadBackendType backend)
+{
+    if (data_ptr == NULL || x == NULL) {
+        return -1;
+    }
+    if (backend == BACKEND_SLEEF) {
+        *(Sleef_quad *)x = *(Sleef_quad *)data_ptr;
+    }
+    else {
+        *(long double *)x = *(long double *)data_ptr;
+    }
+    return 0;
+}
+
+static inline int
+quad_store(char *data_ptr, void *x, QuadBackendType backend)
+{
+    if (data_ptr == NULL || x == NULL) {
+        return -1;
+    }
+    if (backend == BACKEND_SLEEF) {
+        *(Sleef_quad *)data_ptr = *(Sleef_quad *)x;
+    }
+    else {
+        *(long double *)data_ptr = *(long double *)x;
```

Comment:
You can avoid the branching here and in `quad_load` if you split them into separate functions for longdouble and quad. I would expect these functions might get called a lot and repeatedly so avoiding branching should improve performance.